Taste of the World Café is a restaurant that offers a diverse menu and serves generous portions. They debuted a new menu at the start of the year and would like to dig into the customer data to see which menu items are/are not doing well and what the top customers seem to prefer.

This analysis seeks to identify the most and least popular menu items and types of cuisine.

The objectives:
1. Explore the menu_items table to get an idea of what's on the new menu
2. Explore the order_details table to get an idea of the data that's been collected
3. Use both tables to understand how customers are reacting to the new menu

Install dependencies

In [117]:
# %pip install jupysql mysql-connector-python

Initial setup

In [118]:
from dotenv import dotenv_values
from sqlalchemy import create_engine

In [119]:
creds = dotenv_values('creds.env')
user = creds['USER']
passw = creds['PASS']

connection_string = f"mysql+mysqlconnector://{user}:{passw}@localhost/restaurant_db"
engine = create_engine(connection_string)
%reload_ext sql
%sql engine
%config SqlMagic.feedback = False
%config SqlMagic.displaylimit = 1000

Load the SQL file dump

In [120]:
# load the file
#%sql -f data/create_restaurant_db.sql

In [121]:
%%sql

show tables

Tables_in_restaurant_db
menu_items
order_details


### OBJECTIVE 1

View the menu_items table

In [122]:
%%sql
select * from menu_items

menu_item_id,item_name,category,price
101,Hamburger,American,12.95
102,Cheeseburger,American,13.95
103,Hot Dog,American,9.00
104,Veggie Burger,American,10.50
105,Mac & Cheese,American,7.00
106,French Fries,American,7.00
107,Orange Chicken,Asian,16.50
108,Tofu Pad Thai,Asian,14.50
109,Korean Beef Bowl,Asian,17.95
110,Pork Ramen,Asian,17.95


How many items are in the menu?

In [123]:
%%sql

select count(*) 'Menu Items' from menu_items

Menu Items
32


Which menu items are the least and most expensive?

In [124]:
%%sql

(
    select * from menu_items
    order by price desc
    limit 1
)
union all
(
    select * from menu_items
    order by price asc
    limit 1
)


menu_item_id,item_name,category,price
130,Shrimp Scampi,Italian,19.95
113,Edamame,Asian,5.00


What are the least and most expensive Italian dishes? How many Italian dishes are on the menu?

In [125]:
%%sql 

select
    *,
    row_number() over() 'Count'
from menu_items
where category = 'Italian'
order by price
;

menu_item_id,item_name,category,price,Count
124,Spaghetti,Italian,14.50,1
126,Fettuccine Alfredo,Italian,14.50,2
128,Cheese Lasagna,Italian,15.50,3
129,Mushroom Ravioli,Italian,15.50,4
132,Eggplant Parmesan,Italian,16.95,5
125,Spaghetti & Meatballs,Italian,17.95,6
127,Meat Lasagna,Italian,17.95,7
131,Chicken Parmesan,Italian,17.95,8
130,Shrimp Scampi,Italian,19.95,9


There's a total of 9 italian dishes on the menu. Shrimp Scampi being the most expensive and Spaghetti being the least expensive Italian dishes.

How many dishes are in each category? What is the average dish price within each category?

In [126]:
%%sql

select category, count(item_name) 'No. of Dishes',avg(price) 'Average Price' from menu_items
group by category

category,No. of Dishes,Average Price
American,6,10.066667
Asian,8,13.475000
Mexican,9,11.800000
Italian,9,16.750000


With 4 dish categories, it appears the Italian cuisine offers the most expensive dishes on the menu.

### OBJECTIVE 2

View the order_details table

In [127]:
%%sql

select * from order_details
limit 10

order_details_id,order_id,order_date,order_time,item_id
1,1,2023-01-01,11:38:36,109
2,2,2023-01-01,11:57:40,108
3,2,2023-01-01,11:57:40,124
4,2,2023-01-01,11:57:40,117
5,2,2023-01-01,11:57:40,129
6,2,2023-01-01,11:57:40,106
7,3,2023-01-01,12:12:28,117
8,3,2023-01-01,12:12:28,119
9,4,2023-01-01,12:16:31,117
10,5,2023-01-01,12:21:30,117


What is the date range of the table?

In [128]:
%%sql
select count(*),min(order_date),max(order_date) from order_details


count(*),min(order_date),max(order_date)
12234,2023-01-01,2023-03-31


In [129]:
%%sql

select count(distinct(order_date)) 'Date Range (Days)' 
from order_details


Date Range (Days)
90


How many orders were made within this date range?

In [130]:
%%sql

select count(distinct order_id) 'Number of Orders' from order_details

Number of Orders
5370


How many items were ordered within this date range?

In [131]:
%%sql

select count(order_details_id) 'Total Number of Items Ordered' 
from order_details


Total Number of Items Ordered
12234


The data from order_details table shows that there are 12,234 items ordered and 5,370 individual orders made over the course of 3 months.

Which orders had the most number of items?

In [132]:
%%sql

select order_id, count(item_id) as 'Number of Items'
from order_details
group by order_id
order by `Number of Items` desc, order_id desc
limit 20


order_id,Number of Items
4305,14
3473,14
2675,14
1957,14
443,14
440,14
330,14
5200,13
5066,13
4836,13


How many orders had more than 12 items?

In [133]:
%%sql

select order_id, count(item_id) as 'Number of Items', row_number() over() 'COUNT'
from order_details
group by order_id
having `Number of Items` > 12

order_id,Number of Items,COUNT
330,14,1
440,14,2
443,14,3
1274,13,4
1569,13,5
1685,13,6
1734,13,7
1957,14,8
2075,13,9
2126,13,10


### OBJECTIVE 3

Combine the menu_items and order_details tables into a single table

In [134]:
%%sql

select * 
from order_details od
left join menu_items me
    on od.item_id = me.menu_item_id
limit 10


order_details_id,order_id,order_date,order_time,item_id,menu_item_id,item_name,category,price
1,1,2023-01-01,11:38:36,109,109,Korean Beef Bowl,Asian,17.95
2,2,2023-01-01,11:57:40,108,108,Tofu Pad Thai,Asian,14.50
3,2,2023-01-01,11:57:40,124,124,Spaghetti,Italian,14.50
4,2,2023-01-01,11:57:40,117,117,Chicken Burrito,Mexican,12.95
5,2,2023-01-01,11:57:40,129,129,Mushroom Ravioli,Italian,15.50
6,2,2023-01-01,11:57:40,106,106,French Fries,American,7.00
7,3,2023-01-01,12:12:28,117,117,Chicken Burrito,Mexican,12.95
8,3,2023-01-01,12:12:28,119,119,Chicken Torta,Mexican,11.95
9,4,2023-01-01,12:16:31,117,117,Chicken Burrito,Mexican,12.95
10,5,2023-01-01,12:21:30,117,117,Chicken Burrito,Mexican,12.95


What were the least and most ordered items? What categories were they in?

In [135]:
%%sql

(
select item_name, category, count(item_id) 'Number of Orders'
from order_details od
left join menu_items me
    on od.item_id = me.menu_item_id
group by item_name, category
order by `Number of Orders` desc
limit 1
)
union all

(
select item_name, category, count(item_id) 'Number of Orders'
from order_details od
left join menu_items me
    on od.item_id = me.menu_item_id
group by item_name, category
order by `Number of Orders` asc
limit 1,1
)

item_name,category,Number of Orders
Hamburger,American,622
Chicken Tacos,Mexican,123


Hamburgers appears to be the best seller which might be a good reason to stock up more on ingridients for making one. On the other hand, Chicken Tacos seem to be not so popular. It might be worth doing a promo to boost its popularity and sales.

What were the top 5 orders that spent the most money?

In [136]:
%%sql

select order_id, sum(price) 'Total Spent'
from order_details od
left join menu_items me
    on od.item_id = me.menu_item_id
group by order_id
order by `Total Spent` desc
limit 5

order_id,Total Spent
440,192.15
2075,191.05
1957,190.10
330,189.70
2675,185.10


View the details of the highest spend order. Which specific items were purchased?

In [137]:
%%sql

select category, count(item_id) 'Num of Items', sum(price) 'Amount Spent'
from order_details od
left join menu_items me
    on od.item_id = me.menu_item_id
where order_id = 440
group by category


category,Num of Items,Amount Spent
Mexican,2,20.95
American,2,16.00
Italian,8,132.25
Asian,2,22.95


This particular customer seems to prefer Italian dishes but is also open on spending on other cuisines.

View the details of the top 5 highest spend orders

In [138]:
%%sql

select order_id, category, count(item_id) 'Num of Items', sum(price) 'Amount Spent'
from order_details od
left join menu_items me
    on od.item_id = me.menu_item_id
where order_id in (440, 2075, 1957, 330, 2675)
group by order_id, category


order_id,category,Num of Items,Amount Spent
330,Asian,6,87.40
330,American,1,9.00
330,Italian,3,50.40
330,Mexican,4,42.90
440,Mexican,2,20.95
440,American,2,16.00
440,Italian,8,132.25
440,Asian,2,22.95
1957,Asian,3,36.45
1957,American,3,26.50


In [139]:
%%sql

select category, count(item_id) 'Num of Items', sum(price) 'Amount Spent'
from order_details od
left join menu_items me
    on od.item_id = me.menu_item_id
where order_id in (440, 2075, 1957, 330, 2675)
group by category


category,Num of Items,Amount Spent
Asian,17,228.65
American,10,99.35
Italian,26,430.65
Mexican,16,189.45


The Italian dishes seem to be the most popular among the highest spending customers regardless of it being the most expensive items on the menu.